core/lib/src/rocket.rs

```diff
@@ -720,7 +723,8 @@ impl Rocket {
 
     /// Starts the application server and begins listening for and dispatching
     /// requests to mounted routes and catchers. Unless there is an error, this
-    /// function does not return and blocks until program termination.
+    /// function returns `()` and blocks until program termination or graceful
```

Comment:
This should be reworked. Something like "this function does not return unless a shutdown is requested via a `ShutdownHandle`." I will look into whether hyper can return an error at any time other than launch, because that would also be a change in expectations here.

Reply:
Pushed a change. Leaving open with regard to hyper shutting down at other times.